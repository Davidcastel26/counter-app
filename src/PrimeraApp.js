import { Fragment } from "react";
//to request some props as a must use proptype 
import PropTypes from 'prop-types';
// import CounterApp from "./CounterApp";

const PrimeraApp = ( {wave, subTitle} ) =>{
    // -------------- to request some props as a must ------------------------
    // if(!wave){
    //     throw new Error('the wave is has been required')
    // }
    // const saludo = 'Hola Mundo'
    // ----------------------------------------------------------------------
    return(
        <Fragment>
            <h1>{wave}</h1>
            <p>{subTitle}</p>
            {/* <CounterApp value={23}  /> */}
        </Fragment>
    )
}

PrimeraApp.propTypes = {
    wave: PropTypes.string.isRequired,
    subTitle: PropTypes.string
}

PrimeraApp.defaultProps = {
    subTitle: "I wish you the best!",
}

export default PrimeraApp;