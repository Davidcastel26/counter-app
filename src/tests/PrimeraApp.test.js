import React from 'react';
// import { render } from '@testing-library/react';
import PrimeraApp from '../PrimeraApp';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom'

describe('testing the fisrt app component', () => {
    // test("should show the msg 'Hello world'",()=>{
    //     const hey = 'Hello world';
    //     const { getByText } = render( <PrimeraApp wave={hey}/>);
    //     // wrapper.getByText()
    //     expect( getByText(hey)).toBeInTheDocument()
    // })

    test('should show <first app > alrithg', () => {

        const greeting = 'Hello World';
        const wrapper = shallow( <PrimeraApp wave={greeting}/>)

        expect( wrapper ).toMatchSnapshot();

    })

    test('should show the sutitle from props', () => {
        
        const saludo = 'Hello World';
        const subtitulo = "I wish you the best!"
        const wrapper = shallow(
            <PrimeraApp 
                wave={saludo}
                subtitulo={ subtitulo } 
            />
        )

        const textoParrafo = wrapper.find('p').text()
        // document.querySelector()

        expect(textoParrafo).toBe( subtitulo );
        
    })
    

})