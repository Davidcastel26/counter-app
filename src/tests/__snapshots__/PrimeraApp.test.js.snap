// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the fisrt app component should show <first app > alrithg 1`] = `
<Fragment>
  <h1>
    Hello World
  </h1>
  <p>
    I wish you the best!
  </p>
</Fragment>
`;
